crates/rust-analyzer/src/main_loop.rs

```diff
@@ -417,6 +417,8 @@ impl GlobalState {
                 }
             }
 
+            let supports_diagnostic_pull_model = self.config.text_document_diagnostic().is_some();
```

Comment:
This feels a bit confusing. `text_document_diagnostic` actually checks for `related_document_support`. And the latter should probably return `bool`, to avoid situations like #18388?